Tests/Marketplace/marketplace_services.py

```diff
@@ -1641,93 +1665,99 @@ def _enhance_pack_attributes(self, user_metadata, index_folder_path, pack_was_mo
-        )
         self._related_integration_images = self._get_all_pack_images(
             self._displayed_integration_images, displayed_dependencies, dependencies_data,
             pack_dependencies_by_download_count
         )
 
-    def format_metadata(self, user_metadata, index_folder_path, packs_dependencies_mapping, build_number, commit_hash,
-                        pack_was_modified, statistics_handler):
+    def format_metadata(self, index_folder_path, packs_dependencies_mapping, build_number, commit_hash,
+                        pack_was_modified, statistics_handler, pack_names, format_dependencies_only=False):
         """ Re-formats metadata according to marketplace metadata format defined in issue #19786 and writes back
         the result.
 
         Args:
-            user_metadata (dict): user defined pack_metadata, prior the parsing process.
-            public url.
             index_folder_path (str): downloaded index folder directory path.
             packs_dependencies_mapping (dict): all packs dependencies lookup mapping.
             build_number (str): circleCI build number.
             commit_hash (str): current commit hash.
             pack_was_modified (bool): Indicates whether the pack was modified or not.
             statistics_handler (StatisticsHandler): The marketplace statistics handler
-
+            pack_names (set): List of all packs.
+            format_dependencies_only (bool): Indicates whether the metadata formation is just for formatting the dependencies or not.
         Returns:
             bool: True is returned in case metadata file was parsed successfully, otherwise False.
+            bool: True is returned in pack is missing dependencies.
 
         """
         task_status = False
 
         try:
-            self.set_pack_dependencies(user_metadata, packs_dependencies_mapping)
-            if 'displayedImages' not in user_metadata:
-                user_metadata['displayedImages'] = packs_dependencies_mapping.get(
+            self.set_pack_dependencies(self.user_metadata, packs_dependencies_mapping)
+            if 'displayedImages' not in self.user_metadata:
+                self.user_metadata['displayedImages'] = packs_dependencies_mapping.get(
                     self._pack_name, {}).get('displayedImages', [])
                 logging.info(f"Adding auto generated display images for {self._pack_name} pack")
-            dependencies_data = self._load_pack_dependencies(index_folder_path,
-                                                             user_metadata.get('dependencies', {}),
-                                                             user_metadata.get('displayedImages', []))
+            dependencies_data, is_missing_dependencies = \
+                self._load_pack_dependencies(index_folder_path, self.user_metadata.get('dependencies', {}),
+                                             self.user_metadata.get('displayedImages', []), pack_names)
```

Comment:
```suggestion
            dependencies_data, is_missing_dependencies = \
                self._load_pack_dependencies(index_folder_path, pack_names)
```
You can access the attribute in the function itself